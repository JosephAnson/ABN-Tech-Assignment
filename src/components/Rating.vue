<template>
  <div class="flex items-center mb-1 last:mb-0 color-gray-300">
    <span class="i-carbon-star-filled color-yellow mr-1"> </span>
    <span> <slot /></span>
  </div>
</template>
