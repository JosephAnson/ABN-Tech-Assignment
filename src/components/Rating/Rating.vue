<template>
  <div class="flex align-center mb-2 last:mb-0">
    <span class="i-carbon-star color-yellow mr-1"> </span>
    <span class="text-xs"> <slot /></span>
  </div>
</template>
