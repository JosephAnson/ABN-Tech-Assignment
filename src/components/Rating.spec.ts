import { render } from "@testing-library/vue"
import { describe, expect, test } from "vitest"
import Rating from "./Rating.vue"

describe("Rating", () => {
  test("should render rating", () => {
    const { html } = render(Rating, {
      slots: {
        default: "4.7",
      },
    })
    expect(html()).toMatchSnapshot()
  })
})
