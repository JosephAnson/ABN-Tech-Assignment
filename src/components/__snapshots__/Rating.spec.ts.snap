// Vitest Snapshot v1

exports[`Rating > should render rating 1`] = `"<div class=\\"flex items-center mb-1 last:mb-0 color-gray-300\\"><span class=\\"i-carbon-star-filled color-yellow mr-1\\"></span><span>4.7</span></div>"`;
